framework/web/UrlRule.php

```diff
@@ -158,7 +158,8 @@ public function init()
             $this->name = $this->pattern;
         }
 
-        $this->pattern = trim($this->pattern, '/');
+        // don't trim left slashes if more then one slash at the beginning of pattern
+        $this->pattern = preg_replace('/(^(\/)(?!\/)|(\/$))/', '', $this->pattern);
```

Comment:
@samdark, but this solution have a problem: it doesn't include case `////example.com` and `//`. It can be fixed with something like this:

```php
if (substr($this->pattern, 0, 2) === '//' && substr($this->pattern, 2, 1) !== '/' && substr($this->pattern, 2, 1) !== '') {
    $this->pattern = '//' . trim($this->pattern, '/');
} else {
    $this->pattern = trim($this->pattern, '/');
}
```

But I think for now (comparing with the solution above) regex looks more elegant. I can write tests for [new private method](https://github.com/yiisoft/yii2/pull/12697/files#diff-74b1030c748d9da559378bf10fba0078R463) and then we can search for better solution.